aries_cloudagent/holder/routes.py

```diff
@@ -274,6 +329,114 @@ async def credentials_list(request: web.BaseRequest):
+
+    Returns:
+        Verifiable credential record
+
+    """
+    context: AdminRequestContext = request["context"]
+    credential_id = request.match_info["credential_id"]
+
+    session = await context.session()
+    holder = session.inject(VCHolder)
+    try:
+        vc_record = await holder.retrieve_credential_by_given_id(credential_id)
+    except StorageNotFoundError as err:
+        raise web.HTTPNotFound(reason=err.roll_up) from err
+    except StorageError as err:
+        raise web.HTTPBadRequest(reason=err.roll_up) from err
+
+    return web.json_response(vc_record)
+
+
+@docs(
+    tags=["credentials"],
+    summary="Remove DIF credential from wallet by id",
+)
+@match_info_schema(CredIdMatchInfoSchema())
+@response_schema(HolderModuleResponseSchema(), 200, description="")
+async def dif_cred_remove(request: web.BaseRequest):
+    """
+    Request handler for deleting DIF credential.
+
+    Args:
+        request: aiohttp request object
+
+    Returns:
+        Empty production
+
+    """
+    context: AdminRequestContext = request["context"]
+    credential_id = request.match_info["credential_id"]
+
+    session = await context.session()
+    holder = session.inject(VCHolder)
+    try:
+        vc_record = await holder.retrieve_credential_by_given_id(credential_id)
+        await holder.delete_credential(vc_record)
+    except StorageNotFoundError as err:
+        raise web.HTTPNotFound(reason=err.roll_up) from err
+    except StorageError as err:
+        raise web.HTTPBadRequest(reason=err.roll_up) from err
+
+    return web.json_response({})
+
+
+@docs(
+    tags=["credentials"],
+    summary="Fetch DIF credentials from wallet",
+)
+@request_schema(DIFCredentialsListRequestSchema())
+@querystring_schema(CredentialsListQueryStringSchema())
+@response_schema(VCRecordListSchema(), 200, description="")
+async def dif_creds_list(request: web.BaseRequest):
+    """
+    Request handler for searching DIF credential records.
+
+    Args:
+        request: aiohttp request object
+
+    Returns:
+        The credential record list response
+
+    """
+    context: AdminRequestContext = request["context"]
+    session = await context.session()
+    body = await request.json()
+    contexts = body.get("contexts")
+    types = body.get("types")
+    schema_ids = body.get("schema_ids")
+    issuer_id = body.get("issuer_id")
+    subject_id = body.get("subject_id")
+    tag_query = body.get("tag_query")
+    max_results = body.get("max_results")
+
+    holder = session.inject(VCHolder)
+    try:
+        search = holder.search_credentials(
+            contexts, types, schema_ids, issuer_id, subject_id, tag_query
+        )
+        records = await search.fetch(max_results)
+    except StorageNotFoundError as err:
+        raise web.HTTPNotFound(reason=err.roll_up) from err
+    except StorageError as err:
+        raise web.HTTPBadRequest(reason=err.roll_up) from err
+
+    return web.json_response({"results": records})
```

Comment:
`TypeError: Object of type 'VCRecord' is not JSON serializable`

```suggestion
        search = holder.search_credentials(
            contexts, types, schema_ids, issuer_id, subject_id, tag_query
        )
        records = await search.fetch(max_results)
        results = [record.serialize() for record in records]
    except StorageNotFoundError as err:
        raise web.HTTPNotFound(reason=err.roll_up) from err
    except StorageError as err:
        raise web.HTTPBadRequest(reason=err.roll_up) from err

    return web.json_response({"results": results})
```